src/Util.jl

```diff
@@ -6,34 +6,39 @@ tupletypelength(a)=length(a.parameters)
 macro replaceBase(fs...)
```

Comment:
we should probably just get rid of this, since it doesn't make sense anymore.

Reply:
Why doesn't it make sense anymore?  I just tested it out and it seems to work alright.

Reply:
we don't define `log(x::AbstractArray)`

Reply:
or if we do (e.g. `exp(x::Matrix)`) it means something else

Reply:
Part of this update is to override the broadcasting machinery such that `log.(x)` calls `log(x)`; I'm not sure what the right way to do that while still replacing `base` is though.

Reply:
One option would be to give them `v` prefixes, i.e.`vadd`, `vlog`, etc. by default. Then we could have another function which would do the type piracy for us.